src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -149,10 +149,11 @@ CHIP_ERROR BLEManagerImpl::_Init()
 
     memset(mSubscribedConns, 0, sizeof(mSubscribedConns));
 
-    ReturnErrorOnFailure(InitRandomStaticAddress());
     int err = bt_enable(NULL);
     VerifyOrReturnError(err == 0, MapErrorZephyr(err));
 
+    ReturnErrorOnFailure(InitRandomStaticAddress());
```

Comment:
This was done purposefully before `bt_enable` because otherwise the default identity is automatically created using the default rules (which use per-device constant address). See the comment in `InitRandomStaticAddress`. Can you provide a backtrace from the crash? It's not straightforward to generate a new static address using Zephyr's API, so we need to test it carefully.

Reply:
```InitRandomStaticAddress()
  ->bt_addr_le_create_static() - subsys/bluetooth/host/addr.c
      ->create_random_addr()
        ->bt_rand()
          ->lll_csrand_get() - subsys/bluetooth/controller/crypto/crypto.c by default ( not subsys/bluetooth/host/crypto/crypto.c )
            ->entropy_get_entropy(dev_entropy) - dev_entropy is NULL, crashed coz lll_init() called in bt_enable()
```
but not sure should configured by using bt_rand() in file subsys/bluetooth/host/crypto/crypto.c

Reply:
Do you have Zephyr's entropy driver configured/enabled? It looks like you don't have any entropy device configured and that's the issue. You can verify if you're able to call `entropy_get_entropy` in other places. AFAIK the device must be selected with `chosen/zephyr,entropy = &rng;` DTS node.

Reply:
Yes I did and checked, the crashed reason is that dev_entropy is opened by bt_enable(), before bt_enable() dev_entropy is NULL, ```InitRandomStaticAddress``` will crashed in my side

Reply:
OK, gotcha. But if you put `bt_enable()` before `InitRandomStaticAddress()` the latter is not effective anymore, because the default identity cannot be modified once created (at least on our platform). If you look into `InitRandomStaticAddress` you can see that I added an exception for CONFIG_BT_HOST_CRYPTO - perhaps, we can switch to that path for all scenarios if using `bt_addr_le_create_static` before `bt_enable` is undocumented/untested and causes issues.
Btw, which platform/board do you use?

Reply:
CONFIG_BT_HOST_CRYPTO is not configured by default, will try later

nRF52840/pitaya_go (https://wiki.makerdiary.com/pitaya-go/)

Reply:
@Damian-Nordic I still encounter crash both enable BLE and openthread(IEEE802154 radio) on nRF52840, I if disable one of them, it is OK to run matter, but if both enabled, it will crash always in ```radio_nrf5_isr()``` according to zephyr.lst, I will propose another issue in Zephyr project for details

Reply:
ok, my proposal was not to enable BT_HOST_CRYPTO, but to always use the code in InitRandomStaticAddress for BT_HOST_CRYPTO.